components/script/dom/document.rs

```diff
@@ -1975,6 +2047,30 @@ impl Document {
     }
 }
 
+impl VirtualMethods for Document {
+    fn super_type(&self) -> Option<&VirtualMethods> {
+        Some(self.upcast::<Document>() as &VirtualMethods)
+    }
+
+    fn handle_event(&self, event: &Event) {
+        if event.type_() == atom!("keydown") {
+            let key_event = event.downcast::<KeyboardEvent>().unwrap();
```

Comment:
Try to downcast to `KeyboardEvent`, and only then check that its type is `keydown`.